sources.go

```diff
@@ -118,3 +118,14 @@ func (s fakeSource) Discover() ([]string, error) {
 
 	return features, nil
 }
+
+////////////////////////////////////////////////////////////////////////////////
+// Fake Panic Source (used only for testing)
```

Comment:
Could we move fakePanicSource and fakeSource into a separate file (`test_sources.go`)?